core/rio/api/src/main/java/org/eclipse/rdf4j/rio/helpers/XMLWriterSettings.java

```diff
@@ -57,6 +57,17 @@
 	public static final RioSetting<Boolean> USE_SINGLE_QUOTES = new BooleanRioSetting(
 			"org.eclipse.rdf4j.rio.use_single_quotes", "Use single quotes", Boolean.FALSE);
 
+	/**
+	 * Boolean setting for writer to determine whether it should remove the rdf:nodeID attribute from blank nodes.
+	 * <p>
+	 * Defaults to False.
+	 * <p>
+	 * Can be overridden by setting system property {@code org.eclipse.rdf4j.rio.compact_xml}.
+	 */
+	public static final RioSetting<Boolean> COMPACT_XML = new BooleanRioSetting(
+			"org.eclipse.rdf4j.rio.compact_xml",
+			"To generate a compact XML i.e. without Blank Node ID.", Boolean.FALSE);
+
```

Comment:
I just had another look at this, and at the issue, and I realized that this new setting is not really necessary, as there already exists a setting for this: `BasicWriterSettings.INLINE_BLANK_NODES`. 

Why not use that setting to check in RDFXMLPrettyWriter, and remove this setting?

Reply:
Well I didn't quite understand the what this setting INLINE_BLANK_NODES is all about. While understanding the issue I went through the entire link that @barthanssens shared and that contained some test cases that the author was trying to model this problem and I took help from that. But If you're saying that this can be achieved using INLINE_BLANK_NODES settings only then I think I can do that.

Moreover I had a doubt regarding the checks: Is it necessary that the checks will succeed only if @iamaryananand (author of the pull request) will do the commit?

Reply:
> Well I didn't quite understand the what this setting INLINE_BLANK_NODES is all about. While understanding the issue I went through the entire link that @barthanssens shared and that contained some test cases that the author was trying to model this problem and I took help from that. But If you're saying that this can be achieved using INLINE_BLANK_NODES settings only then I think I can do that.

Not a problem, it took me a while to remember this as well. But yes, I believe we can reuse the existing setting.

> Moreover I had a doubt regarding the checks: Is it necessary that the checks will succeed only if @iamaryananand (author of the pull request) will do the commit?

If you mean the sign off check: it doesn't matter who created the pull request, it looks at the commits inside the pull request. Each individual commit that you add needs to be signed off by the author of that commit, and that author needs to be on record as having the signed the Eclipse Contributor Agreement. So if you add a commit, _you_ need to sign it off, if @iamaryananand adds a commit, they need to sign off that commit. 

If you are interested in learning why we need this: our project is part of the Eclipse Foundation, which requires a formal record of each contribution, who has created it, and that that person has given us license to use their contribution. As an open-source contributor you still own your own code, so we need written evidence that you given us permission to reuse it. In other words, we need to have an audit trail for [Intellectual Property (IP)](https://www.eclipse.org/projects/handbook/#ip).

Reply:
Okay! So we managed to delete the CompactXML setting as you requested and are using the existing INLINE_BLANK_NODES for our purpose.